content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md

```diff
@@ -123,19 +123,43 @@ by using a list of well known domain socket paths. To use different container ru
 if there are more than one installed on the provisioned node, specify the `--cri-socket`
 argument to `kubeadm init`. See [Installing runtime](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-runtime).
 1. (Optional) Unless otherwise specified, kubeadm uses the network interface associated
-with the default gateway to advertise the master's IP. To use a different
+with the default gateway to advertise the control-plane's IP. To use a different
 network interface, specify the `--apiserver-advertise-address=<ip-address>` argument
 to `kubeadm init`. To deploy an IPv6 Kubernetes cluster using IPv6 addressing, you
 must specify an IPv6 address, for example `--apiserver-advertise-address=fd00::101`
 1. (Optional) Run `kubeadm config images pull` prior to `kubeadm init` to verify
-connectivity to gcr.io registries.   
+connectivity to gcr.io registries.
 
-Now run:
+To initialize the control-plane node run:
 
 ```bash
 kubeadm init <args>
 ```
 
+### Considerations about ControlPlaneEndpoint
+
+Instead of using `--apiserver-advertise-address`, `kubeadm init` also allows you to use the `--control-plane-endpoint` flag.
+While `--apiserver-advertise-address` can be used to set the advertise address for this particular control-plane node's API
+server, `--control-plane-endpoint` is preferred and can be used to set the shared endpoint for all control-plane nodes.
+Such an endpoint can be either a DNS name or an IP address of a load-balancer.
+
+If you have plans to upgrade this single control-plane kubeadm cluster to high-availability you should definitely use
+`--control-plane-endpoint`. This will allow you to pass any DNS name and later change the address that the DNS name resolves to.
+Use your preferred solution for mapping this node's IP address to a DNS name. For example:
```

Comment:
If you don't have DNS name available, but you want to preserve the possibility to turn your cluster into a highly available cluster, as a workaround you can add an entry in `/etc/hosts` on all your nodes.

Reply:
@detiber mentioned that `/etc/hosts` is not a good recommendation.

Reply:
@detiber mentioned that /etc/hosts is not a good recommendation.

Reply:
The main reason that I'd like to avoid mentioning /etc/hosts is because if it is explicitly in the docs then someone will configure it without understanding the implications and will run into issues.

Reply:
I'm ok in removing the reference to /etc/hosts  and replacing “, as a workaround you can add an entry in /etc/hosts on all your nodes." with ", please contact your network administrator to evaluate possible workarounds" or something similar